api/main_endpoints/routes/User.js

```diff
@@ -406,4 +408,104 @@ router.post('/getUserById', async (req, res) => {
   });
 });
 
+router.get('/isUserSubscribed', (req, res) => {
+  User.findOne({ email: req.query.email }, function(error, result) {
+    if (error) {
+      res.status(BAD_REQUEST).send({ message: 'Bad Request.' });
+    }
+
+    if (!result) {
+      return res
+        .status(NOT_FOUND)
+        .send({ message: `${req.query.email} not found.` });
+    }
+    if (result.emailOptIn)
+      return res.status(OK).send({ message: 'Subscribed!' });
+    else return res.status(OK).send({ message: 'No!' });
+  });
+});
+
+router.post('/setUserEmailPreference', (req, res) => {
+  const email = req.body.email;
+  const emailOptIn = req.body.emailOptIn;
+  User.updateOne(
+    { email: email },
+    { emailOptIn: emailOptIn },
+    function(error, result) {
+      if (error) {
+        res.status(BAD_REQUEST).send({ message: 'Bad Request.' });
+      }
+
+      if (result.n === 0) {
+        return res
+          .status(NOT_FOUND)
+          .send({ message: `${email} not found.` });
+      }
+      return res.status(OK).send({
+        message: `${email} was updated.`,
+        emailOptIn: emailOptIn,
+      });
+    }
+  );
+});
+
+router.post('/getUserDataByEmail', (req, res) => {
```

Comment:
what is this api for?

Reply:
this is for the front end to get the user first and last name so that it can be displayed on the front end

Reply:
i think the only thing we need to return is their name and if their opted in?